### PR TITLE
use protobuf 1.2.1

### PR DESCRIPTION
**Pull request by siddontang**
@BusyJay @hhkbp2 @overvenus

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -11,15 +11,15 @@ dependencies = [
  "crc 1.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "fs2 0.4.0 (registry+https://github.com/rust-lang/crates.io-index)",
  "getopts 0.2.14 (registry+https://github.com/rust-lang/crates.io-index)",
- "kvproto 0.0.1 (git+https://github.com/pingcap/kvproto)",
+ "kvproto 0.0.1 (git+https://github.com/pingcap/kvproto?branch=siddontang/protobuf-1.2.1)",
```
Comment (hhkbp2) [inline Cargo.lock]: Why not master branch?

**Comment by hhkbp2** [comment]
Rest LGTM

File: Cargo.lock (hunk 1 above)
Comment (siddontang) [inline Cargo.lock]: kvproto must use 1.2.1, so here I have to use the branch now. 

After kvproto and tipb PR merged, I will use the master branch.

**Comment by ngaut** [comment]
LGTM

**Comment by ngaut** [comment]
Older version rust can't compile protobuf-1.2.1

**Comment by hhkbp2** [comment]
LGTM

**Comment by siddontang** [comment]
The latest protobuf and gRPC uses `?` instead of `try!`, I file an issue https://github.com/stepancheg/rust-protobuf/issues/203. 

But ultimately, we must fix the core dump with the newer rust version.

**Comment by siddontang** [comment]
PTAL @BusyJay @hhkbp2

**Comment by siddontang** [comment]
I will merge the kvproto and tipb first, use the master and then update the cargo later.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**